### PR TITLE
Add logging with io.Writer

### PR DESCRIPTION
**Pull request by davidwyld**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -9,11 +9,15 @@ package main
...
 import (
 	"fmt"
+	"io"
 	"log"
 	"net/http"
 	"os"
+	"time"
 )
 
+var b io.Writer
```
Comment (drj11) [inline main.go]: Can we call this `log` instead of `b`, please?

File: main.go [hunk 2]
```diff
@@ -22,11 +26,21 @@ func apiHandler(w http.ResponseWriter, r *http.Request) {
...
 	dateTime := r.FormValue("dateTime")
-	fmt.Printf("apikey %s, product %s, version %s, uuid %s, eventID %s, dateTime %s, path %s\n",
+	fmt.Fprintf(b, "apikey %s, product %s, version %s, uuid %s, eventID %s, dateTime %s, path %s\n",
 		apikey, product, version, uuid, eventID, dateTime, path)
 }
 
 func main() {
+	// creates a new log file with a timestamped name
+	t := time.Now()
+	fname := "ingalog_" + t.Format("20060102150405") + ".log"
```
Comment (drj11) [inline main.go]: If we use `.log` as an extension, we don't need `log` in the first part.
So, replace `ingalog_` with `inga_`.

**Review by drj11 — CHANGES_REQUESTED** [review]
Looks good.

There are kleverer ways to introduce the `io.Writer` variable for the log, but that's really after-school homework. I like the simplicity.

We need to do some research to satisfy ourselves that this is safe (as in thread-safe) and doesn't result in partially written or torn log entries.

File: main.go (hunk 1 above)
Comment (davidwyld) [reply main.go]: done

File: main.go (hunk 2 above)
Comment (davidwyld) [reply main.go]: yup.

**Pull request merged**